src/ngraph/op/scatter_elements_update.cpp

```diff
@@ -0,0 +1,118 @@
+    element::Type axis_et = get_input_element_type(3);
+
+    const PartialShape& data_shape = get_input_partial_shape(0);
+    const PartialShape& indices_shape = get_input_partial_shape(1);
+    const PartialShape& updates_shape = get_input_partial_shape(2);
+    const PartialShape& axis_shape = get_input_partial_shape(3);
+
+    NODE_VALIDATION_CHECK(this,
+                          indices_et.is_integral(),
+                          "Indices element type must be integral_number, but is: ",
+                          indices_et);
+
+    NODE_VALIDATION_CHECK(this,
+                          axis_et.is_integral(),
+                          "Axis element type must be integral_number, but is: ",
+                          axis_et);
+
+    NODE_VALIDATION_CHECK(this,
+                          data_et == updates_et,
+                          "Data type and updates type are required to be the same. ",
+                          "Got: ",
+                          data_et,
+                          " and: ",
+                          updates_et);
+
+    NODE_VALIDATION_CHECK(this,
+                          axis_shape.compatible(PartialShape{}) ||
+                              axis_shape.compatible(PartialShape{1}),
+                          "Axis input shape are required to be scalar or 1D tensor. ",
+                          "Got: ",
+                          axis_shape);
+
+    NODE_VALIDATION_CHECK(this,
+                          indices_shape.compatible(updates_shape),
+                          "Indices and updates input shapes are required to be equal. ",
+                          "Got: ",
+                          indices_shape,
+                          " and: ",
+                          updates_shape);
+
+    NODE_VALIDATION_CHECK(this,
+                          indices_shape.rank().compatible(data_shape.rank()),
+                          "Indices rank and data rank are required to be equal. ",
+                          "Got: ",
+                          indices_shape.rank(),
+                          " and: ",
+                          data_shape.rank());
+
+    if (input_value(3).get_node_shared_ptr()->is_constant() && data_shape.rank().is_static())
+    {
+        const auto axis_input = as_type_ptr<op::v0::Constant>(input_value(3).get_node_shared_ptr());
+        auto axis = axis_input->cast_vector<int64_t>().at(0);
+
+        const auto data_rank_length = data_shape.rank().get_length();
+        NODE_VALIDATION_CHECK(
+            this,
+            (-data_rank_length < axis) && (axis < data_rank_length - 1),
+            "Axis value has to be in range [-r, r-1] where r is rank of data shape. ",
+            "Data rank: ",
+            data_rank_length,
+            "Got axis value: ",
+            axis);
+    }
+
+    set_output_size(1);
+    set_output_type(0, data_et, data_shape);
+}
+
+shared_ptr<Node> op::v3::ScatterElementsUpdate::copy_with_new_args(const NodeVector& new_args) const
+{
```

Comment:
`copy_with_new_args` is now deprecated. Use `clone_with_new_inputs` which takes an `OutputVector` as its input.

Reply:
Thanks. Should I just left `check_new_args_count(this, new_args);` ? 
I couldn't find any helper function to check inputs size like this so I've created the check here for now.

Reply:
That's fine.